clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -86,6 +85,25 @@ public AbortedTransaction(long pid, long firstOffset) {
         }
 
         @Override
+        public boolean equals(Object o) {
+            if (this == o)
+                return true;
+            if (o == null || getClass() != o.getClass())
+                return false;
+
+            AbortedTransaction that = (AbortedTransaction) o;
+
+            return pid == that.pid && firstOffset == that.firstOffset;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = (int) (pid ^ (pid >>> 32));
```

Comment:
Unfortunately `Objects.hash` is a lot more expensive because it boxes the primitives and also allocates an array. It may not matter here, but we generally try to make `hashCode` and `equals` allocation free, so I think I'll leave as is.